Remotion/Web/ClientScript/Scripts/ListMenu.ts

```diff
@@ -14,39 +14,97 @@
+var _listMenu_listMenuInfos: Dictionary<ListMenu_MenuInfo> = {}
 
-var _listMenu_itemClassName = 'listMenuItem';
-var _listMenu_itemFocusClassName = 'listMenuItemFocus';
-var _listMenu_itemDisabledClassName = 'listMenuItemDisabled';
-var _listMenu_requiredSelectionAny = 0;
-var _listMenu_requiredSelectionExactlyOne = 1;
-var _listMenu_requiredSelectionOneOrMore = 2;
+var _listMenu_itemClassName: string = 'listMenuItem';
+var _listMenu_itemFocusClassName: string = 'listMenuItemFocus';
+var _listMenu_itemDisabledClassName: string = 'listMenuItemDisabled';
+var _listMenu_requiredSelectionAny: number = 0;
+var _listMenu_requiredSelectionExactlyOne: number = 1;
+var _listMenu_requiredSelectionOneOrMore: number = 2;
 
-function ListMenu_MenuInfo(id, itemInfos)
+class ListMenu_MenuInfo
 {
-  this.ID = id;
-  this.ItemInfos = itemInfos;
+  private _id: string;
+  private _itemInfos: ListMenu_ItemInfo[];
+
+  public get ID(): string { return this._id; }
+  public get ItemInfos(): ListMenu_ItemInfo[] { return this._itemInfos; }
+
+  constructor(id: string, itemInfos: ListMenu_ItemInfo[])
+  {
+    this._id = id;
+    this._itemInfos = itemInfos;
+  }
 }
 
-function ListMenuItemInfo (id, category, text, icon, iconDisabled, requiredSelection, isDisabled, href, target, diagnosticMetadata, diagnosticMetadataForCommand)
+class ListMenu_ItemInfo
 {
-  this.ID = id;
-  this.Category = category;
-  this.Text = text;
-  this.Icon = icon;
-  this.IconDisabled = iconDisabled;
-  this.RequiredSelection = requiredSelection;
-  this.IsDisabled = isDisabled;
-  this.Href = href;
-  this.Target = target;
-  this.DiagnosticMetadata = diagnosticMetadata;
-  this.DiagnosticMetadataForCommand = diagnosticMetadataForCommand;
+  private readonly _id: string;
+  private readonly _category: string;
+  private readonly _text: string;
+  private readonly _icon: string;
+  private readonly _iconDisabled: string;
+  private readonly _requiredSelection: number;
+  private readonly _isDisabled: boolean;
+  private readonly _href: string;
+  private readonly _target: string;
+  private readonly _diagnosticMetadata: Dictionary<string | boolean>;
+  private readonly _diagnosticMetadataForCommand: Dictionary<string | boolean>;
+
+  public get ID(): string { return this._id }
+  public get Category(): string { return this._category; }
+  public get Text(): string { return this._text; }
+  public get Icon(): string { return this._icon; }
+  public get IconDisabled(): string { return this._iconDisabled; }
+  public get RequiredSelection(): number { return this._requiredSelection; }
+  public get IsDisabled(): boolean { return this._isDisabled; }
+  public get Href(): string { return this._href; }
+  public get Target(): string { return this._target; }
+  public get DiagnosticMetadata(): Dictionary<string | boolean> { return this._diagnosticMetadata; }
+  public get DiagnosticMetadataForCommand(): Dictionary<string | boolean> { return this._diagnosticMetadataForCommand; }
+
+  constructor(
```

Comment:
I asked MK and he is okay with the TypeScript shortcut syntax for quick assigning constructor parameters:
```ts
constructor(
  public readonly id,
  public readonly category,
  ... ) { }
```

If the constructor sets additional state then the normal variant of declaring and assigning each field should be used

Reply:
Reasoning for this is to keep all state declarations at one place so we can get a quick and complete overview of a class's state declarations.